internal/controller/components/dashboard/dashboard_controller_kustomize_test.go

```diff
@@ -0,0 +1,260 @@
+// Black-box tests that run as dashboard_test package and validate exported/public functions
+package dashboard_test
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/opendatahub-io/opendatahub-operator/v2/api/common"
+	dashboardctrl "github.com/opendatahub-io/opendatahub-operator/v2/internal/controller/components/dashboard"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster/gvk"
+	odhtypes "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/utils/test/fakeclient"
+)
+
+// setupTempDirWithParams creates a temporary directory with the proper structure and params.env file.
+func setupTempDirWithParams(t *testing.T) string {
```

Comment:
To do correctly unit test on `ApplyParams`, we probably need to pass the fs as args (or in some other way), so that we can mock it in memory instead of writing in fs. What do you think about?